File: examples/read.js
// example reads single adc input
// run:
// > node read.js

"use strict";

const PORT = 1; // i2c-1
const RAWDATA = false;

const adc = require("../ads1115");

adc.read(
    PORT,
    adc.ADR_48, // address 0x48
    adc.MUX_I0_GND, // AIN0 - GND
    adc.GAIN_4096, // 4096 mV
    adc.RATE_128, // 128 SPS
    RAWDATA, // rawdata ?
    function(data) {
        if (data === undefined)
            console.log(adc.error_text());
        else
            console.log(data, "mV");
    });

File: cpp/read.cpp
/*
 * example reads single adc input and prints value
 *
 * build:
 * > make
 *
  * run:
 * > ./read
 *
 * read.cpp
 *
 */

#include <stdio.h>

#include "ads1115_lib.h"

#define PORT 1
#define RAWDATA false

int main()
{
    double value;

    bool ret = ads1115::read(
        PORT,
        ads1115::ADR_48,
        ads1115::MUX_I0_GND,
        ads1115::GAIN_4096,
        ads1115::RATE_128,
        RAWDATA,
        value);

    if (ret)
        printf("%.0fmV\n", value);
    else
        printf("%s\n", ads1115::error_text());

    return 0;
}


